### PR TITLE
refactor(KtModal): Convert to Composition API, Add Types, Fix Margins

### PR DESCRIPTION
**Pull request by FlorianWendelborn**

### CODE REVIEW

File: packages/kotti-ui/source/kotti-modal/KtModal.vue [hunk 1]
```diff
@@ -18,37 +18,105 @@
...
+.modal-mask {
+	position: fixed;
+	top: 0;
+	left: 0;
+	z-index: $zindex-4;
+	display: flex;
+	width: 100%;
+	height: 100%;
+	background-color: rgba(0, 0, 0, 0.46);
+	transition: opacity 0.3s ease;
```
Comment (carsoli) [inline packages/kotti-ui/source/kotti-modal/KtModal.vue:58]: variable for short transition or whatever is shortest?

File: packages/kotti-ui/source/kotti-modal/KtModal.vue [hunk 2]
```diff
@@ -18,37 +18,105 @@
...
+
+<style lang="scss">
+@import '../kotti-style/_variables.scss';
+
+.modal-mask {
+	position: fixed;
+	top: 0;
+	left: 0;
+	z-index: $zindex-4;
+	display: flex;
```
Comment (carsoli) [inline packages/kotti-ui/source/kotti-modal/KtModal.vue:54]: display: flex feels useless, nothing else is set, does it make sense to you

File: packages/kotti-ui/source/kotti-modal/KtModal.vue [hunk 3]
```diff
@@ -18,37 +18,105 @@
...
+	opacity: 0;
+}
+
+.modal-leave-active {
+	opacity: 0;
+}
+
+.modal-enter .modal-container,
+.modal-leave-active .modal-container {
+	-webkit-transform: scale(1.1);
```
Comment (carsoli) [inline packages/kotti-ui/source/kotti-modal/KtModal.vue:119]: is it needed or already added by that plugin you mentioned before that adds the webkit/moz specific attributes? (can't remember what it's called sorry)

File: packages/kotti-ui/source/kotti-modal/KtModal.vue [hunk 4]
```diff
@@ -18,37 +18,105 @@
...
-		},
-	},
-}
+})
 </script>
+
+<style lang="scss">
+@import '../kotti-style/_variables.scss';
+
+.modal-mask {
```
Comment (carsoli) [inline packages/kotti-ui/source/kotti-modal/KtModal.vue:49]: can u make the `modal` be the wrapper and use `&-` for practically all of them?

File: packages/kotti-ui/source/kotti-modal/KtModal.vue [hunk 5]
```diff
@@ -18,37 +18,105 @@
...
-	methods: {
-		closeModal() {
-			if (this.closeOutside) this.$emit('close')
-		},
-	},
-}
+})
 </script>
+
+<style lang="scss">
```
Comment (carsoli) [inline packages/kotti-ui/source/kotti-modal/KtModal.vue:46]: can we add `scoped` or is this about stacking context or what?
as in did you intentionally not so the styles are global?

File: packages/kotti-ui/source/kotti-modal/KtModal.vue (hunk 3 above)
Comment (FlorianWendelborn) [reply packages/kotti-ui/source/kotti-modal/KtModal.vue:119]: Should be handled by `autoprefixer`, yes

**Review by carsoli — APPROVED** [review]

File: packages/kotti-ui/source/kotti-modal/KtModal.vue (hunk 5 above)
Comment (FlorianWendelborn) [reply packages/kotti-ui/source/kotti-modal/KtModal.vue:46]: It’s not `scoped` as `yoda` is currently using those styles

File: packages/kotti-ui/source/kotti-modal/KtModal.vue (hunk 4 above)
Comment (FlorianWendelborn) [reply packages/kotti-ui/source/kotti-modal/KtModal.vue:49]: Sure

File: packages/kotti-ui/source/kotti-modal/KtModal.vue (hunk 2 above)
Comment (FlorianWendelborn) [reply packages/kotti-ui/source/kotti-modal/KtModal.vue:54]: The child uses it via `align-self`

**Pull request merged**